### ISSUE TITLE
生成古诗如同随机文本的问题

### ISSUE THREAD

**Issue by Imposingapple** [dbiir/UER-py#127]
您好，我使用UER的GPT-2预训练方法训练了一个古诗的模型，然后做预测的时候发现生成的就好像是随机的文本，有时甚至还有很多[UNK]，想请教下这是为什么？
![2](https://user-images.githubusercontent.com/39848377/113375497-13235600-93a2-11eb-85db-42337d68ab4d.JPEG)
我的输入是“床前明月光，”

------------------------------------------------------------------------------------------------------------------
我的数据是nlp_chinese_corpus这个仓库中的中国诗词，我按照book_review.txt的形式，一行放了一首诗，然后按照您给的GPT-2预训练示例的输入指令进行预处理与预训练，我的数据文本大概是这样子的
![1](https://user-images.githubusercontent.com/39848377/113375228-8b3d4c00-93a1-11eb-963f-4062d0a2a19b.JPEG)

预训练的指令如下：
CUDA_VISIBLE_DEVICES=1 python3 pretrain.py --dataset_path datasets/poems.pt --vocab_path models/google_zh_vocab.txt --output_model_path models/poems_model.bin --config_path models/gpt2/config.json --learning_rate 1e-4 --world_size 1 --gpu_ranks 0 --tie_weight --embedding word_pos --remove_embedding_layernorm --encoder transformer --mask causal --layernorm_positioning pre --target lm

测试的指令如下：
python generate_lm.py --load_model_path ../models/poems_model.bin-100000 --vocab_path ../models/google_zh_vocab.txt --test_path ../corpora/test_poems.txt --prediction_path ../corpora/predicted.txt --config_path ../models/gpt2/config.json

**Comment by Embedding** [comment]
只有一张卡并且只训练了10万步。可能训练不充分。建议训练更多的步数。
关于[UNK]问题，我们使用了扩展的词典，细节请参考这里[这里](https://huggingface.co/uer/gpt2-chinese-poem)。

**Comment by Imposingapple** [comment]
您好，我参看了您给的链接，然后完全运行了示范给定的代码，并按照预训练的选项运行了generate_lm.py，效果虽然比之前好了不少，没了[UNK]，但结果貌似还是不尽如人意，包括很多[SEP]字符，输入五言可是生成的全是七言（我的数据集七言和五言一起输入进去的，是nlp_chinese_corpus的诗，一行一首，七言不会比五言多很多的），而且一开始的[CLS]不知道为什么在输出时就是[cl##s]

这是我的测试代码：
![sh](https://user-images.githubusercontent.com/39848377/113546406-7de0c580-961e-11eb-9a3f-94758589eac5.JPEG)

这是我的生成结果：
![result](https://user-images.githubusercontent.com/39848377/113546419-846f3d00-961e-11eb-8fbc-3e27c1e05e92.JPEG)

这是和我的数据点差不多的数据点的运行状态：
![state](https://user-images.githubusercontent.com/39848377/113546505-b6809f00-961e-11eb-9ac7-6e32b84c9492.JPEG)

我的vocab用的是您给的链接里下载下来的，并命名为poem_zh_vocab.txt

请问这是什么原因？
还有想请教一下[CLS]是在程序里哪里加的，我看了UER的dataloader部分好像没有找到？

求大佬指教，谢谢！

**Comment by hhou435** [comment]
您好，我们刚刚更新了generate_lm.py，在开头默认加上了[CLS]，不需要手动添加了。
此外，我们也在Modelzoo中提供了GPT2古诗预训练模型。
https://github.com/dbiir/UER-py/wiki/%E9%A2%84%E8%AE%AD%E7%BB%83%E6%A8%A1%E5%9E%8B%E4%BB%93%E5%BA%93#%E5%9F%BA%E4%BA%8E%E4%B8%8D%E5%90%8C%E8%AF%AD%E6%96%99%E7%9A%84%E4%B8%AD%E6%96%87gpt-2%E9%A2%84%E8%AE%AD%E7%BB%83%E6%A8%A1%E5%9E%8B
您可以下载后与您现在训练的模型比较一下生成结果。
出现[SEP]代表了生成结束。
